onnxruntime/test/platform/threadpool_test.cc

```diff
@@ -137,6 +137,30 @@ void TestBurstScheduling(const std::string& name, int num_tasks) {
   }
 }
 
+void TestPoolCreation(const std::string&, int iter) {
+  // Test creating and destroying thread pools.  This can be used with Valgrind to help
+  // check for memory leaks related to the initialization and clean-up code.  For instance
+  //
+  //  valgrind --leak-check=full ./onnxruntime_test_all --gtest_filter=ThreadPoolTest.TestPoolCreation_10Iter
+  //
+  // We create #iter thread pools, and within each of them run a loop of #per_iter steps.
+  std::atomic<std::ptrdiff_t> ctr{0};
+  constexpr std::ptrdiff_t per_iter = 1024;
+  constexpr int num_threads = 4;
+  for (auto i = 0; i < iter; i++) {
+    auto tp = onnxruntime::make_unique<ThreadPool>(&onnxruntime::Env::Default(),
+                                                   onnxruntime::ThreadOptions(),
+                                                   nullptr,
+                                                   num_threads,
+                                                   true);
+    tp->ParallelFor(per_iter, 0.0,
```

Comment:
So,  100 iterations only take 12 ms, it used to be very slow, about a few seconds per iteration. I'm curious how it get improved.

Reply:
Interesting, 0.1ms is about what I would expect here.  The work in creating and destroying a thread pool in tests like these should just be a few system calls, perhaps ~1 MB allocation, and then synchronization between the threads followed by their exit.  

If we do see several-seconds on examples like this then that's a bug we need to identify and fix.  Is that currently occurring?

Reply:
Never mind. I was wrong.